### PR TITLE
34 create generic modal system

### LINKED ISSUE #34: Create generic modal system
Modals should be able to display

- Title
- Description
- Arbitrary number of buttons (Ok, yes, no, cancel, etc)
- Arbitrary number of text fields with names and placeholders

The goal is to use this wherever a popup is needed

### PR DESCRIPTION
**Pull request by Sciman101**

### CODE REVIEW

File: src/routes/towers.js [hunk 1]
```diff
@@ -21,10 +44,11 @@ export const Towers = observer((props) => {
                        }}
                        onJoinTowerSelected={() => {
                            // TODO implement cool modals or something
-                           const code = prompt("Enter the code for the tower you want to join");
+                           /*const code = prompt("Enter the code for the tower you want to join");
                            if (towerId !== null) {
                                joinTower(code);
-                           }
+                           }*/
```
Comment (IncPlusPlus) [inline src/routes/towers.js]: Should the TODO and commented code be removed now?

File: src/components/TowerDetails.js [hunk 2]
```diff
@@ -24,17 +29,24 @@ export const TowerDetails = observer(function TowerDetails(props) {
     // Called when the 'invite people' button is pressed
     const inviteHandler = () => {
         // Get an invite code from the tower
+        setInviteButtonState(INVITE_WAITING);
         generateInviteCode(props.tower.id).then(invite => {
             navigator.clipboard.writeText(invite.inviteCode);
-            alert("Invite copied to clipboard! This will expire in one hour");
+            setInviteButtonState(INVITE_COPIED);
+            //alert("Invite copied to clipboard! This will expire in one hour");
```
Comment (IncPlusPlus) [inline src/components/TowerDetails.js]: Should this alert still be here?

**Review by IncPlusPlus — CHANGES_REQUESTED** [review]
Surprisingly elegant. Using `react-modal` was a good call. I just have two questions that I'd like answered and then I'll slap a green check on this.

File: src/components/TowerDetails.js (hunk 2 above)
Comment (Sciman101) [reply src/components/TowerDetails.js]: Good call, ill remove that

**Comment by IncPlusPlus** [comment]
Also, the modal can get _really_ funky if I have the dev tools open. idk if this is something you want to bother fixing right now though. 
image

File: src/routes/towers.js (hunk 1 above)
Comment (Sciman101) [reply src/routes/towers.js]: Will remove

**Review by IncPlusPlus — APPROVED** [review]

**Pull request merged**